GitUI/UserControls/RevisionGrid/FilterInfo.cs

```diff
@@ -423,31 +423,27 @@ private void GetBranchRevisionFilter(ArgumentBuilder filter)
                 // All commits
                 filter.Add("--reflog");
             }
-            else if (IsShowCurrentBranchOnlyChecked || (IsShowFilteredBranchesChecked && !string.IsNullOrWhiteSpace(BranchFilter)))
+            else if (IsShowCurrentBranchOnlyChecked && !string.IsNullOrWhiteSpace(currentBranch.Value))
             {
-                // See handling for --all below, explicitly add most recent stash
-                if (AppSettings.ShowStashes)
-                {
-                    // stash@{0} requires that the repo has commits and --glob=refs/stash is handled as refs/stash/*,
-                    // so this weird pattern must be used: "--glob=refs/stas[h]"
-                    filter.Add($"--glob={GitRefName.RefsStashPrefix.Remove(GitRefName.RefsStashPrefix.Length - 1, 1)}[h]");
-                }
+                // Git default, no option by default (stashes is special).
+                // Add as filter anyway as the branch must exist and the repo must contain commits
+                // otherwise Git will exit with errors.
```

Comment:
Ditto here

Reply:
That line would be 136 chars, too long in my opinion.
Rewrote that too (merged the comment below), which would have resulted in a 176 char comment.